Remove always-nil error returns

tklauser opened the pull request:
These errors are always nil, so they can be omitted and get rid of the respective error checking code. Found using [`unparam`](https://github.com/mvdan/unparam). Note that we cannot enable the `unparam` linter yet in `golangci-lint` because there are some false positives.

mhofstetter reviewed (approved):
Thanks!

Artyop commented on daemon/cmd/deletion_queue.go:
```diff
@@ -69,10 +66,10 @@ func newDeletionQueue(lc cell.Lifecycle, p promise.Promise[*Daemon]) *deletionQu
+func (dq *deletionQueue) lock(ctx context.Context) {
 	if err := os.MkdirAll(defaults.DeleteQueueDir, 0755); err != nil {
 		log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueDir).Error("Failed to ensure CNI deletion queue directory exists")
-		return nil
```
I'm curious about this case, why not return the error created by `os.MkdirAll` ?

mhofstetter replied:
probably because the whole lock is best-effort (at least it's handled this way in the rest of the `lock` method) :thinking:

tklauser replied:
Thanks, good call. The operations following `(*deletionQueue).lock` in `(*deletionQueue).Start` would anyway fail if `os.MkdirAll` fails and the directory `defaults.DeleteQueueDir` doesn't exist.

I'll change to return the error instead in this case.

tklauser replied:
> probably because the whole lock is best-effort (at least it's handled this way in the rest of the lock method) 🤔

Yeah, all of this is best-effort but I guess we will spam the log less with successive errors in case we return the `os.MkdirAll` error.

Artyop reviewed (approved):
Just a question for my own curiosity, otherwise lgtm

mhofstetter commented on daemon/cmd/deletion_queue.go:
```diff
@@ -69,10 +66,10 @@ func newDeletionQueue(lc cell.Lifecycle, p promise.Promise[*Daemon]) *deletionQu
+		return
 	}
 
 	var err error
```
nit: error handling in this code could be cleaned up a bit more. still related to error handling (~ context of this PR) - but i can also create a follow up PR

* early return (with removal of explicit `else` case)
* don't reuse error variable in second condition (while locking)

```
	var err error
	dq.lf, err = lockfile.NewLockfile(defaults.DeleteQueueLockfile)
	if err != nil {
		log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueLockfile).Warn("Failed to lock queued deletion directory, proceeding anyways. This may cause CNI deletions to be missed.")
		return
	}

	if err := dq.lf.Lock(ctx, true); err != nil {
		log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueLockfile).Warn("Failed to lock queued 	deletion directory, proceeding anyways. This may cause CNI deletions to be missed.")
		dq.lf.Close()
		dq.lf = nil
	}
}
```

tklauser replied:
How about the following incremental diff:

```diff
diff --git a/daemon/cmd/deletion_queue.go b/daemon/cmd/deletion_queue.go
index 83a13b982bd0..3724b4a01e43 100644
--- a/daemon/cmd/deletion_queue.go
+++ b/daemon/cmd/deletion_queue.go
@@ -72,20 +72,24 @@ func newDeletionQueue(lc cell.Lifecycle, p promise.Promise[*Daemon]) *deletionQu
 func (dq *deletionQueue) lock(ctx context.Context) error {
        if err := os.MkdirAll(defaults.DeleteQueueDir, 0755); err != nil {
                log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueDir).Error("Failed to ensure CNI deletion queue directory exists")
+               // Return error to avoid attempting successive df.processQueuedDeletes accessing
+               // defaults.DeleteQueueLockfile and erroring because of the non-existent directory.
                return err
        }
 
+       // Don't a non-nil return error from here on so successive dq.processQueuedDeletes can still
+       // continue with best effort.
        var err error
        dq.lf, err = lockfile.NewLockfile(defaults.DeleteQueueLockfile)
        if err != nil {
                log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueLockfile).Warn("Failed to lock queued deletion directory, proceeding anyways. This may cause CNI deletions to be missed.")
-       } else {
-               err = dq.lf.Lock(ctx, true)
-               if err != nil {
-                       log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueLockfile).Warn("Failed to lock queued deletion directory, proceeding anyways. This may cause CNI deletions to be missed.")
-                       dq.lf.Close()
-                       dq.lf = nil
-               }
+               return nil
+       }
+
+       if err = dq.lf.Lock(ctx, true); err != nil {
+               log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueLockfile).Warn("Failed to lock queued deletion directory, proceeding anyways. This may cause CNI deletions to be missed.")
+               dq.lf.Close()
+               dq.lf = nil
        }
        return nil
 }
```

(folding in the error return on `os.MkdirAll` failure).

mhofstetter replied:
if we return the error from `MkdirAll` there's no need to log the error in the `lock` function as it gets logged in `Start` (maybe worth to wrap the returning error with `fmt.Errorf` to give some more context - e.g. adding the path - even though no longer as structured log field and part of the error message).

Otherwise :+1:

tklauser replied:
Good point. I think for consistency I'm going to drop the log after the call to `dq.lock` so all error logging is in a single place within `dq.lock`.

tklauser committed:
Remove always-nil error returns

These errors are always nil, so they can be omitted and get rid of the
respective error checking code. Found using unparam. Note that we cannot
enable the unparam linter yet in golangci-lint because there are some
false positives.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

giorio94 reviewed (approved):
Looks good for my codeowners, thanks!

aanm commented on daemon/cmd/deletion_queue.go:
```diff
@@ -44,7 +44,6 @@ func (dq *deletionQueue) Start(cell.HookContext) error {
 		if err := dq.lock(d.ctx); err != nil {
-			log.WithError(err).Error("deletionQueue: lock failed")
 			return
 		}
```
should we remove the if statement?

tklauser replied:
No, we still want to return in case of `os.MkdirAll` error and avoid calling successive operations which require the directory to exist to avoid flodding the log with warnings. See previous discussion in https://github.com/cilium/cilium/pull/36910#pullrequestreview-2539339534 and https://github.com/cilium/cilium/pull/36910#pullrequestreview-2539357242 for why the code looks as is now.

pippolo84 reviewed (approved):
Thanks!
